shpc/main/modules/base.py

```diff
@@ -475,8 +456,47 @@ def install(
             name = "%s:%s" % (name, tag.name)
         logger.info("Module %s was created." % name)
 
-        # Install the module (symlink) to the view and create version file
-        if view:
-            view.install(module_dir)
-
         return container_path
+
+    def view_install(self, view_name, name, tag=None, force=False):
```

Comment:
I’ll need to think this over - I like the separate functions but I don’t like the redundancy - that makes it harder to maintain and more likely the two become out of sync.

Reply:
Yep, noticed the redundancy too. And it's not just `install` and `view_install`: there are several other places (esp. in this module: `uninstall`, `check`) where we go from a module name to the container dir/path, module dir, etc, possibly with a namespace.

Reply:
Yes! So perhaps uninstall could use this shared function to?

Reply:
Yep, especially as I've got a commit (not pushed yet) to create `view_uninstall` from the current `uninstall` and mirror `install`/`view_install`